lib/dependency_validator.dart

```diff
@@ -57,6 +63,29 @@ void run() {
 
   logger.info('Validating dependencies for $packageName\n');
 
+  // `loadPackageConfig` (used below) relies on having a file built from an absolute URI.
+  final packageConfig = File.fromUri(Uri.file('${Directory.current.path}/.dart_tool/package_config.json'));
+
+  // URIs in packageConfig may be relative to its directory. Change the working directory to account for this.
+  final rootDir = Directory.current;
+  final dir = Directory('${rootDir.path}/.dart_tool');
+  Directory.current = dir;
+
+  // Find packages that provide executables.
+  final packagesWithExecutables = Set<String>();
+  for (final package in (await loadPackageConfig(packageConfig)).packages) {
```

Comment:
There is a separate `package_config_discovery.dart` entry point that provides APIs for discovering and parsing the package config file. We should be able to use [`findPackageConfig(Directory.current)`](https://pub.dev/documentation/package_config/latest/package_config.package_config_discovery/findPackageConfig.html) to replace most of this logic.